hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/OMDBInsightSearchEndpoint.java

```diff
@@ -0,0 +1,390 @@
+
+      // Split the path to ensure it's at least at the bucket level
+      String[] pathComponents = startPrefix.split("/");
+      if (pathComponents.length < 3 || pathComponents[2].isEmpty()) {
+        return createBadRequestResponse(
+            "Invalid startPrefix: Path must be at the bucket level or deeper.");
+      }
+
+      // Ensure the limit is non-negative
+      limit = Math.max(0, limit);
+
+      // Initialize response object
+      KeyInsightInfoResponse insightResponse = new KeyInsightInfoResponse();
+      long replicatedTotal = 0;
+      long unreplicatedTotal = 0;
+      boolean keysFound = false; // Flag to track if any keys are found
+      String lastKey = null;
+
+      // Search for non-fso keys in KeyTable
+      Table<String, OmKeyInfo> openKeyTable =
+          omMetadataManager.getOpenKeyTable(BucketLayout.LEGACY);
+      Map<String, OmKeyInfo> obsKeys =
+          retrieveKeysFromTable(openKeyTable, startPrefix, limit, prevKey);
+      for (Map.Entry<String, OmKeyInfo> entry : obsKeys.entrySet()) {
+        keysFound = true;
+        KeyEntityInfo keyEntityInfo =
+            createKeyEntityInfoFromOmKeyInfo(entry.getKey(), entry.getValue());
+        insightResponse.getNonFSOKeyInfoList()
+            .add(keyEntityInfo); // Add to non-FSO list
+        replicatedTotal += entry.getValue().getReplicatedSize();
+        unreplicatedTotal += entry.getValue().getDataSize();
+        lastKey = entry.getKey(); // Update lastKey
+      }
+
+      // Search for fso keys in FileTable
+      Map<String, OmKeyInfo> fsoKeys = searchOpenKeysInFSO(startPrefix, limit, prevKey);
+      for (Map.Entry<String, OmKeyInfo> entry : fsoKeys.entrySet()) {
+        keysFound = true;
+        KeyEntityInfo keyEntityInfo =
+            createKeyEntityInfoFromOmKeyInfo(entry.getKey(), entry.getValue());
+        insightResponse.getFsoKeyInfoList()
+            .add(keyEntityInfo); // Add to FSO list
+        replicatedTotal += entry.getValue().getReplicatedSize();
+        unreplicatedTotal += entry.getValue().getDataSize();
+        lastKey = entry.getKey(); // Update lastKey
+      }
+
+      // If no keys were found, return a response indicating that no keys matched
+      if (!keysFound) {
+        return noMatchedKeysResponse(startPrefix);
+      }
+
+      // Set the aggregated totals in the response
+      insightResponse.setReplicatedDataSize(replicatedTotal);
+      insightResponse.setUnreplicatedDataSize(unreplicatedTotal);
+      insightResponse.setLastKey(lastKey);
+
+      // Return the response with the matched keys and their data sizes
+      return Response.ok(insightResponse).build();
+    } catch (IOException e) {
+      // Handle IO exceptions and return an internal server error response
+      return createInternalServerErrorResponse(
+          "Error searching open keys in OM DB: " + e.getMessage());
+    } catch (IllegalArgumentException e) {
```

Comment:
In what use case, it will throw IllegalArgumentException, pls check, I have a doubt, and even if this method throws in some scenario, then pls write javadoc comment.

Reply:
@dombizita 

The `IllegalArgumentException` in the given code can occur due to invalid arguments passed to methods. In the context of the `convertToObjectPath` method and the search logic, possible scenarios include:

1. **Invalid Volume or Bucket Names:**
   - The `validateNames` method can throw an `IllegalArgumentException` if volume or bucket names do not meet required criteria.

2. **Invalid Path Format:**
   - If the provided path does not adhere to the expected format, such as missing necessary components (volume, bucket, directory), it can lead to an `IllegalArgumentException`.

3. **Conversion Failures:**
   - When converting path components to IDs, if the format is incorrect or if the conversion logic encounters unexpected values, it might throw an `IllegalArgumentException`.